### ISSUE TITLE
[BUG] ORC write does not include full timestamp metrics which causes spark to not do predicate push down

### ISSUE THREAD

**Issue by revans2** [rapidsai/cudf#13899]
**Describe the bug**
I am not totally sure if this is a bug or a new feature, but it ends up being a performance problem for Spark.

ORC statistics for timestamps are confusing. They include a minimum, maximum, minimum UTC, maximum UTC, and then min and max nanoseconds.

CUDF only sets minimum and maximum UTC. Spark and java ORC by default use only minimum and maximum +min/max nanos for predicate push down.  This results in predicate push down not working at all for timestamps in ORC.

https://github.com/NVIDIA/spark-rapids/pull/9068

The Spark CPU writer fills in all of the fields. It would really be great if CUDF could do the same. The nano seconds is probably not that important, but timestamp really is important.

**Steps/Code to reproduce bug**
Write a file out with CUDF in ORC. Try to read the data in with spark along with a filter where the value is outside of the range for any value in a timestamp column.

This will result in all of the data being read when CUDF wrote the file, but if the CPU wrote the file none of it would have been read.

**Expected behavior**
Predicate push down works no matter who wrote the data.

**Comment by vuule** [comment]
@revans2 could you share more info about the nanosecond fields? I don't see anything about that in the specs.
I sill add min/max in https://github.com/rapidsai/cudf/pull/13848; IIUC, the change is trivial because we always write files in UTC.

**Comment by revans2** [comment]
@vuule I don't know about the spec. i was just reading through the code and looking at the dump of the metrics form the command line tool.

https://github.com/apache/orc/blob/a9e49656558e54f3f5c1b1eb0b9d70e2ad80bc21/proto/orc_proto.proto#L70-L72

is the protocol buffer code for the nanoseconds. If they are not provided, then the default for min is 0, and the default for max is 999999. The only reason I mentioned it is because when I dumped the footer metrics I saw that the max for our file had the 999999 at the end for nanoseconds and I wanted to understand why it was different from what Spark had written out. I don't think it would change anything functionally
